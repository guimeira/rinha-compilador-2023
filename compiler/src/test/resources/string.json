{"name":"string.rinha","expression":{"kind":"Let","name":{"text":"a","location":{"start":4,"end":5,"filename":"string.rinha"}},"value":{"kind":"Str","value":"str","location":{"start":8,"end":13,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":19,"end":20,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"a","location":{"start":29,"end":30,"filename":"string.rinha"}},"location":{"start":23,"end":31,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"b","location":{"start":37,"end":38,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Int","value":3,"location":{"start":41,"end":42,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Var","text":"a","location":{"start":45,"end":46,"filename":"string.rinha"}},"location":{"start":41,"end":46,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":52,"end":53,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"b","location":{"start":62,"end":63,"filename":"string.rinha"}},"location":{"start":56,"end":64,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"c","location":{"start":70,"end":71,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":74,"end":75,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Int","value":3,"location":{"start":78,"end":79,"filename":"string.rinha"}},"location":{"start":74,"end":79,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":85,"end":86,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"c","location":{"start":95,"end":96,"filename":"string.rinha"}},"location":{"start":89,"end":97,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"d","location":{"start":103,"end":104,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":107,"end":108,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Str","value":"bla","location":{"start":111,"end":116,"filename":"string.rinha"}},"location":{"start":107,"end":116,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":122,"end":123,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"d","location":{"start":132,"end":133,"filename":"string.rinha"}},"location":{"start":126,"end":134,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"e","location":{"start":140,"end":141,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Str","value":"bla","location":{"start":144,"end":149,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Var","text":"a","location":{"start":152,"end":153,"filename":"string.rinha"}},"location":{"start":144,"end":153,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":159,"end":160,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"e","location":{"start":169,"end":170,"filename":"string.rinha"}},"location":{"start":163,"end":171,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"f","location":{"start":177,"end":178,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":181,"end":182,"filename":"string.rinha"}},"op":"Mul","rhs":{"kind":"Int","value":3,"location":{"start":185,"end":186,"filename":"string.rinha"}},"location":{"start":181,"end":186,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":192,"end":193,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"f","location":{"start":202,"end":203,"filename":"string.rinha"}},"location":{"start":196,"end":204,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"g","location":{"start":210,"end":211,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Int","value":3,"location":{"start":214,"end":215,"filename":"string.rinha"}},"op":"Mul","rhs":{"kind":"Var","text":"a","location":{"start":218,"end":219,"filename":"string.rinha"}},"location":{"start":214,"end":219,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":225,"end":226,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"g","location":{"start":235,"end":236,"filename":"string.rinha"}},"location":{"start":229,"end":237,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"tt","location":{"start":243,"end":245,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Int","value":1,"location":{"start":248,"end":249,"filename":"string.rinha"}},"op":"Eq","rhs":{"kind":"Int","value":1,"location":{"start":253,"end":254,"filename":"string.rinha"}},"location":{"start":248,"end":254,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"h","location":{"start":260,"end":261,"filename":"string.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":264,"end":265,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Var","text":"tt","location":{"start":268,"end":270,"filename":"string.rinha"}},"location":{"start":264,"end":270,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":276,"end":277,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"h","location":{"start":286,"end":287,"filename":"string.rinha"}},"location":{"start":280,"end":288,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"i","location":{"start":294,"end":295,"filename":"string.rinha"}},"value":{"kind":"Tuple","first":{"kind":"Int","value":1,"location":{"start":299,"end":300,"filename":"string.rinha"}},"second":{"kind":"Str","value":"abc","location":{"start":301,"end":306,"filename":"string.rinha"}},"location":{"start":298,"end":307,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":313,"end":314,"filename":"string.rinha"}},"value":{"kind":"Print","value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":323,"end":324,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Var","text":"i","location":{"start":327,"end":328,"filename":"string.rinha"}},"location":{"start":323,"end":328,"filename":"string.rinha"}},"location":{"start":317,"end":329,"filename":"string.rinha"}},"next":{"kind":"Let","name":{"text":"j","location":{"start":335,"end":336,"filename":"string.rinha"}},"value":{"kind":"Function","parameters":[{"text":"x","location":{"start":342,"end":343,"filename":"string.rinha"}}],"value":{"kind":"Var","text":"x","location":{"start":350,"end":351,"filename":"string.rinha"}},"location":{"start":339,"end":353,"filename":"string.rinha"}},"next":{"kind":"Print","value":{"kind":"Binary","lhs":{"kind":"Var","text":"a","location":{"start":361,"end":362,"filename":"string.rinha"}},"op":"Add","rhs":{"kind":"Var","text":"j","location":{"start":365,"end":366,"filename":"string.rinha"}},"location":{"start":361,"end":366,"filename":"string.rinha"}},"location":{"start":355,"end":367,"filename":"string.rinha"}},"location":{"start":331,"end":367,"filename":"string.rinha"}},"location":{"start":309,"end":367,"filename":"string.rinha"}},"location":{"start":290,"end":367,"filename":"string.rinha"}},"location":{"start":272,"end":367,"filename":"string.rinha"}},"location":{"start":256,"end":367,"filename":"string.rinha"}},"location":{"start":239,"end":367,"filename":"string.rinha"}},"location":{"start":221,"end":367,"filename":"string.rinha"}},"location":{"start":206,"end":367,"filename":"string.rinha"}},"location":{"start":188,"end":367,"filename":"string.rinha"}},"location":{"start":173,"end":367,"filename":"string.rinha"}},"location":{"start":155,"end":367,"filename":"string.rinha"}},"location":{"start":136,"end":367,"filename":"string.rinha"}},"location":{"start":118,"end":367,"filename":"string.rinha"}},"location":{"start":99,"end":367,"filename":"string.rinha"}},"location":{"start":81,"end":367,"filename":"string.rinha"}},"location":{"start":66,"end":367,"filename":"string.rinha"}},"location":{"start":48,"end":367,"filename":"string.rinha"}},"location":{"start":33,"end":367,"filename":"string.rinha"}},"location":{"start":15,"end":367,"filename":"string.rinha"}},"location":{"start":0,"end":367,"filename":"string.rinha"}},"location":{"start":0,"end":367,"filename":"string.rinha"}}
