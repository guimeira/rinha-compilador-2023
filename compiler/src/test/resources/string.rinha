let a = "str";
let _ = print(a);
let b = 3 + a;
let _ = print(b);
let c = a + 3;
let _ = print(c);
let d = a + "bla";
let _ = print(d);
let e = "bla" + a;
let _ = print(e);
let f = a * 3;
let _ = print(f);
let g = 3 * a;
let _ = print(g);
let tt = 1 == 1;
let h = a + tt;
let _ = print(h);
let i = (1,"abc");
let _ = print(a + i);
let j = fn(x) => { x };
print(a + j)
