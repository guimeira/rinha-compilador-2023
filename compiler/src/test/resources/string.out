str
3str
str3
strbla
blastr
strstrstr
strstrstr
strtrue
str(1,abc)
str<#closure>

